src/fits.cc

```diff
@@ -439,7 +439,12 @@ std::string makeErrorMessage(std::string const &fileName, int status, std::strin
     }
     os << "\ncfitsio error stack:\n";
     char cfitsioMsg[FLEN_ERRMSG];
+    // fits_read_errmsg can return a junk string with control characters
     while (fits_read_errmsg(cfitsioMsg) != 0) {
+        cfitsioMsg[FLEN_ERRMSG-1] = char(0); // ensure termination
+        std::size_t len=strlen(cfitsioMsg);
+        for(std::size_t i=0;i<len;i++)
+            if(cfitsioMsg[i] < char(32)) cfitsioMsg[i] = '.';
```

Comment:
(I think my previous comment got rebased out).

I think `isprint()` as the test would be more explicit.

Reply:
Yes, ignore the previous version. isprint is clearly better.